### ISSUE TITLE
37110570: Need a way to get access to calendar/contact.app push certificates

### ISSUE THREAD

**Issue by openradar-mirror** [lionheart/openradar-mirror#19283]
#### Description

As per https://support.apple.com/en-us/HT208312 Apple is going to deprecate the macOS Server calendar and contact servers soon. People will need to switch to other products (e.g. those mentioned on the support page, or new ones).

Loosing calendar and contacts in macOS Server is unfortunate, but kinda ok as the products already support open standards (CalDAV/CardDAV) and stuff can be replaced w/ 3rd party products.
However there is one key facility which mac&iOS are going to loose: calendar and contact push.
That is: 3rd party servers (/users of those) have no way acquire an Apple push certificate for calendar/contacts by any means, and are hence unable to push to those applications.

The result is going to be a highly degraded user experience for calendar/contact users, as the servers can't ensure the client database is proactively kept in sync.
This results in a large increase of sync conflicts as the users databases are more likely to be out of date.
It also results in the calendar agent having to *poll* a lot, thus greatly affecting battery life of iOS devices.

In summary we need a way for developers or users to acquire calendar push certificates.
Or alternatively, a different (official) mechanism to initiate a calendar server refresh (e.g. one option could be a custom app of the server developer which receives its regular push, and then has an API/entitlement to make the calendar API refresh).

FWIW, even Apple's own calendar server project is affected by this issue: https://github.com/apple/ccs-calendarserver/issues/508#issuecomment-362223227

-
Product Version: 
Created: 2018-02-01T11:23:07.042280
Originated: 2018-02-01T00:00:00
Open Radar Link: http://www.openradar.me/37110570
